src/irmin-git/irmin_git_intf.ml

```diff
@@ -112,7 +115,7 @@ module type Sigs = sig
 
   type reference = Reference.t [@@deriving irmin]
 
-  module Content_addressable (G : Git.S) : sig
+  module Content_addressable (G : Git.S) (I : Irmin.Info.S) : sig
```

Comment:
I'm no expert in irmin-git and git, but I don't this that the commit infos can be customized in that context.

Would it be possible for your feature to avoid all the diffs in `irmin-git` and `irmin-mirage`?

Reply:
I don't really like the parameterization of the `irmin-git` package either; it feels heavy and a bit out of place from the other parameters. I first attempted adding a custom info to `irmin-git` to encapsulate the changes but ran into the lack of the unix dependency.

I see two possible ways to avoid such a heavy interface change.

1. Extend `Irmin_git.G` to support customizing either an entire info or just the handling of dates and add a corresponding maker. This would remove the info parameterization, but I'm not sure how much it would reduce changes internally to `irmin-git` without writing some code.
2. Don't worry about formatting timestamps and just let `Irmin.Info` print the integer value. The PR would then be only the [last commit](https://github.com/mirage/irmin/pull/1836/commits/e9278a091b847a320400d156ba1bcbeb4d4a0227), minus the changes to `irmin-unix`. 

Do you have a preference?

Reply:
Just to see what it would look/feel like, I went ahead and wrote some code to rough out the first idea. It still needs to touch almost all the same files but it does remove the info from being a top-level parameter. This would essentially replace the first commit in this PR, and I believe the second commit could be easily modified to fit into this scheme. Here is a diff (https://github.com/mirage/irmin/compare/main...metanivek:extend-git-store) if you want to take a look for comparison.

Reply:
I unfortunately prefer solution 2 because modifying the irmin-git signatures in such a way also means updating many mirage unikernels. We could think about shipping that diff at the same time as another functor refactoring.

For the time being, the last commit is the only one that can be easily merged

Reply:
Got it. Yeah, even extending `G` results in having to touch outside usages of the package. Not a big deal though! Nicely formatted dates definitely don't justify that kind of ecosystem impact. 😄 

I'll close this PR (to keep the code/conversation as archive) and open a new one with just the pretty-printers. Thanks for the insights and help!

Reply:
Thank you :)